### PR TITLE
mach: Upgrade `cargo-deny` and report unknown errors in `test-tidy`

### LINKED ISSUE #38945: deny.toml missing duplicate checks
#38914 added two new duplicate crates, `objc2-metal` and `objc2-quartz-core` and neither were added to deny.toml. However the duplicate check isn't showing issues with this. Is there a problem with the duplicate check using deny.toml?

### PR DESCRIPTION
**Pull request by mukilan**
The versions of `cargo-deny` older than 0.18.6 have a bug which causes the executions of `cargo-deny check` to prematurely fail when reading the advisory db (https://github.com/EmbarkStudios/cargo-deny/issues/804).

This error is ignored by `test-tidy` since the error message doesn't have the expected JSON fields, causing `test-tidy` to succeed even when there are valid issues in `deny.toml` or `Cargo.lock`.

So upgrade the `cargo-deny` version  installed by `mach` to be the latest version and ensure that at least the version with the fix is installed on the system. Also fix the `test-tidy` code to always fail when the exit code from `cargo-deny` is non-zero.

This patch also updates `deny.toml` to include exceptions to allow `./mach test-tidy` to pass. Some of these need to be investigated separately from this change.

Fixes #41845.
Fixes #38945.

Testing: Tested locally on NixOS.

### CODE REVIEW

**Comment by servo-highfive** [comment]
Heads up! This PR modifies the following files:
 * @jschwe: deny.toml

File: python/servo/platform/base.py [hunk 1]
```diff
@@ -95,13 +95,13 @@ def cargo_deny_installed() -> bool:
...
             (major, minor, micro) = result.stdout.strip().split(" ")[1].split(".", 2)
-            return (int(major), int(minor), int(micro)) >= (0, 18, 1)
+            return (int(major), int(minor), int(micro)) >= (0, 18, 6)
 
         if cargo_deny_installed():
             return False
 
         print(" * Installing cargo-deny...")
-        if subprocess.call(["cargo", "install", "cargo-deny@0.18.3", "--locked"]) != 0:
+        if subprocess.call(["cargo", "install", "cargo-deny@0.19.0", "--locked"]) != 0:
```
Comment (jschwe) [inline python/servo/platform/base.py:104]: Maybe we should also perform a version check before running `cargo-deny` as part of test-tidy, so people are notified that they need to upgrade cargo deny?

Comment (mukilan) [reply python/servo/platform/base.py:104]: Not sure if that is something we need to handle in code - the CI runs will anyway use the newer cargo-deny and should report issues in the PR. Perhaps a message on Zulip would be sufficient to notify active developers?

Comment (jschwe) [reply python/servo/platform/base.py:104]: Yeah, but people would be confused if `test-tidy` is passing locally, but failing in CI. Having an error message would be helpful, especially for contributors who only occasionally contribute, they may miss such a message on zulip.

**Review by jschwe — APPROVED** [review]

File: python/servo/platform/base.py (hunk 1 above)
Comment (mukilan) [reply python/servo/platform/base.py:104]: That seems like a useful feature to have in general as I think there were instances in the past where we've recommended folks to try upgrading to newer cargo-deny for various reasons. I think that is better handled in a separate change as we probably want to think about the design and do it cleanly e.g, avoid duplicating code for version checks between bootstrap and `test-tidy`. I'd keep this patch to the minimum change required to fix the CI from allowing bad PRs to land.

Comment (jschwe) [reply python/servo/platform/base.py:104]: sounds reasonable.
